darling/traps.c

```diff
@@ -384,7 +384,9 @@ static void darling_ipc_inherit(task_t old_task, task_t new_task)
 }
 
 // No vma from 4.11
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0)
+static vm_fault_t mach_mmap_fault(struct vm_fault *vmf)
```

Comment:
Is there a reason not to use `vm_fault_t` on older kernels?

Reply:
@bugaevc I think the `vm_fault_t` is only available in newer kernels from the digging i did. I am not a C/C++ developer nor do I really hack on the kernel, but I did find the following commits in the kernel that speak about this new return type

https://github.com/torvalds/linux/commit/1c8f422059ae5da07db7406ab916203f9417e396#diff-60c20ebcb0698e4c5701e55b99a7004c

https://github.com/torvalds/linux/commit/b3ec9f33acb8d3a6173515d3d547be969dc70566#diff-60c20ebcb0698e4c5701e55b99a7004c

I just know using the conditional macro keeps the module compiling and older and newer kernels and leaves in the original condition for kernels >= 4.11

If you or anyone has any additional insight or recommendations i would gladly make changes and test them.

Reply:
Ah, this makes sense :+1: